ink-wrapper/src/codegen.rs

```diff
@@ -333,6 +353,25 @@ fn message_args(args: &[MessageParamSpec<PortableForm>], metadata: &InkProject)
     }
 }
 
+/// Generates an event definition as a variant in the `Event` enum.
+///
+/// Note that these definitions are hidden in a module to avoid name clashes (just in case someone uses `Event` as a
+/// type name), so references to types defined in the contract need to be prefixed with `super::`.
+fn define_event(event: &EventSpec<PortableForm>, metadata: &InkProject) -> rust::Tokens {
+    quote! {
+        $(docs(event.docs()))
+        $(event.label()) {
+            $(for field in event.args() {
+                $(docs(field.docs()))
+                $(field.label()): $(if resolve(metadata, field.ty().ty().id()).is_custom() { super:: })
+                    $(type_ref(field.ty().ty().id(), metadata)),
```

Comment:
It took me a while to realize what's going on here:
1. `$(field.label())` is event's argument's name
2. the second part is its type with `super::` prefix if it's a _custom_ type (i.e. defined in the contract)
3. the line break is probably applied by `cargo fmt` and has nothing to do with a line break in the resulting file

Is it not possible to extract these to an intermediate variable and reuse? Something like
```rust
$(for field in event.args() {
  let arg_name = field.label();
  let arg_type = ...;

  $(docs(field.docs()))
  $(arg_name): $(arg_type),

  $[ '\n' ]
}
```

Reply:
Sure, I'll try to work something out. The thing you posted doesn't work out of the box, because `let arg_name` would be injected into the generated source in this case, which is not what we want, but I think something similar should work.